### PR TITLE
Reworks the turbine to be an actual engine

### PR DESCRIPTION
**Pull request by SapphicOverload**
## About The Pull Request

This PR allows the turbine to provide thrust like a real engine while it's running (equivalent to around 4 plasma thrusters while running at maximum power) and fixes a few things like being able to make power from nothing and disconnecting when docking or undocking.

Also updates the sprites to make it nicer to look at.

260364787-c289bee8-8389-48de-a6da-29b305bed64d
Sprites are from BeeStation/BeeStation-Hornet#7634, by PestoVerde322 and PacifistDalek

## Why It's Good For The Game

For how much worse and less efficient the turbine is at power generation compared to the TEG, I thought it deserved a secondary use to make it more interesting. This also allows for the possibility of a new ship that uses one (or more!) as its main engine, which would be cool.

Also turbines violating the laws of thermodynamics to make power from nowhere is maybe bad or something.

## Changelog

:cl: sapphicoverload, PestoVerde322, PacifistDalek
add: turbines now provide thrust for ships
tweak: turbines respect conservation of energy instead of moving gas around for free
tweak: turbine control console now shows the internal pressure of the turbine
code: refactored how gas flows through the turbine
fix: fixed turbines making free energy from nowhere at low RPM
fix: fixed turbines forcing air into the output turf when it's blocked
fix: fixed turbines not updating air at the input and output turfs
fix: fixed turbines disconnecting when docking and undocking
imageadd: new turbine sprites
/:cl:

### CODE REVIEW

**Review by ToasterBiome — APPROVED** [review]
Looks good. Wait, this isn't my repo.

**Comment by Apogee-dev** [comment]
I'm rubbing my dirty little mapper paws together criminally at the thought of mapping with this thing

**Review by rye-rice — APPROVED** [review]
new turbine is good
Fun fact: the old turbine sprite is from 2004!
image

**Comment by Latentish** [comment]
Turbine with a real sprite is cursed as all hell and I love it.

**Review by MarkSuckerberg — APPROVED** [review]
seems good. needs a testmerge though of course
one day I hope all engines require this much thought

File: code/game/machinery/shuttle/turbine.dm [hunk 1]
```diff
@@ -159,105 +182,153 @@
...
+// Return the maximum power output
+/obj/machinery/power/shuttle/engine/turbine/return_fuel_cap()
+	return ((COMPFRICTION*(compressor ? compressor.efficiency : 1) / (TURBGENQ*4))**TURBGENG) * TURBGENQ * productivity
+
+// Return the maximum power output
+/obj/machinery/power/shuttle/engine/turbine/update_engine()
+	if(!(flags_1 & INITIALIZED_1))
+		return FALSE
+	thruster_active = TRUE
+	if(panel_open)
```
Comment (west3436) [inline code/game/machinery/shuttle/turbine.dm]: `if(panel_open || !compressor)`

File: code/game/machinery/shuttle/turbine.dm [hunk 2]
```diff
@@ -108,49 +124,56 @@
...
+/obj/machinery/power/compressor/process(delta_time)
+	return
 
-	inturf.transfer_air_ratio(gas_contained, 0.1)
+/obj/machinery/power/compressor/process_atmos(delta_time)
+	// RPM function to include compression friction - be advised that too low/high of a compfriction value can make things screwy
+	rpm -= 1
+	rpm = (0.9 * rpm) + (0.1 * rpmtarget)
+	rpm = min(rpm, (COMPFRICTION*efficiency)/2)
+	rpm = max(0, rpm - (rpm**2)/(COMPFRICTION*efficiency))
```
Comment (west3436) [inline code/game/machinery/shuttle/turbine.dm:166]: Not necessarily requesting a change here but ensure `efficiency` isn't ever equal to 0

**Comment by thgvr** [comment]
@tmtmtl30 review?

**Comment by thgvr** [comment]
I didn't notice this failed to merge so LGTM

**Pull request merged**